### PR TITLE
3539 add module adjustments

### LINKED ISSUE #3539: RPC `add-module` not working as expected
The following test to add a new module to an existing definition fails to produce the expected result.

The test is based on a simple transition system between data constructors `a` to `f`  (see `test-a-to-f`).
Transitions defined in the original `TEST` module are:
```
      a
     / \
    V   V
    b   c
        |
        V
        d
```

A `NEW` module is added using `add-module` to add the following transitions:

```     d
        |
        V
        e
        |
        V
        f
```
1) `add-module` request with `NEW` module
2) `execute` request with start state `c` and `module: NEW` parameter

Expected: After adding the `NEW` module, an `execute` request starting in state `c` should proceed to `f` before getting stuck.

Observed: the `execute` request gets _stuck_ in state `c`

Note that with the original `TEST` module the `execute` request would at least transition to `d` before getting stuck. It appears as though the `import TEST` of the `NEW` module is ignored.

Test data:
[Kore definition: module-addition.kore.txt](https://github.com/runtimeverification/haskell-backend/files/10928052/module-addition.kore.txt)
[add-module-request.json.txt](https://github.com/runtimeverification/haskell-backend/files/10928054/add-module-request.json.txt)
[exec-from-c-request.json.txt](https://github.com/runtimeverification/haskell-backend/files/10928053/exec-from-c.json.txt)

### PR DESCRIPTION
**Pull request by jberthold**
Fixes #3539

### Scope:

* modifies the `add-module` request to not include a `name` field and contain just the module syntax (`module BLAH ... endmodule []`

* Modifies the module loading to (transitively) consider rules that were imported into the main module instead of defined there directly.
  This change was performance-tested using KEVM

### Estimate:

---

###### Review checklist

The author performs the actions on the checklist. The reviewer evaluates the work and checks the boxes as they are completed.

-   [ ] **Summary.** Write a summary of the changes. Explain what you did to fix the issue, and why you did it. Present the changes in a logical order. Instead of writing a summary in the pull request, you may push a clean Git history.
-   [ ] **Documentation.** Write documentation for new functions. Update documentation for functions that changed, or complete documentation where it is missing.
-   [ ] **Tests.** Write unit tests for every change. Write the unit tests that were missing before the changes. Include any examples from the reported issue as integration tests.
-   [ ] **Clean up.** The changes are already clean. Clean up anything near the changes that you noticed while working. This does not mean only spatially near the changes, but logically near: any code that interacts with the changes!

### CODE REVIEW

**Comment by jberthold** [comment]
> * This change needs to be thoroughly tested  (incl. a performance comparison)

Tested with KEVM proofs (`make test-prove-smoke`), speed and success rate was unchanged.

File: docs/2022-07-18-JSON-RPC-Server-API.md [hunk 1]
```diff
@@ -333,18 +333,37 @@ Same as for execute
...
 
-* `name` is the name of the main module.
-* `module` is module represented in KORE.
+* `module` is a module represented in textual KORE format. The module may import modules that have been loaded or added earlier.
+
+### Error Response:
+
+If the textual KORE in `module` is syntactically wrong, the response will use the error code
+`Could not parse pattern` with the parse error in the `data` field (also see below).
+
```
Comment (jberthold) [inline docs/2022-07-18-JSON-RPC-Server-API.md:347]: @tothtamas28 could you please have a look at the modified description?
The request type has changed (`name` field omitted, parsing a _single_ module now), so this is what I made of your pull request #3535

**Review by ehildenb — APPROVED** [review]

File: docs/2022-07-18-JSON-RPC-Server-API.md (hunk 1 above)
Comment (tothtamas28) [reply docs/2022-07-18-JSON-RPC-Server-API.md:347]: Looks good to me.

**Pull request merged**